libcaf_core/caf/byte.hpp

```diff
@@ -1,73 +1 @@
-// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
-// the main distribution directory for license terms and copyright or visit
-// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
-
-#include <cstdint>
-#include <type_traits>
-
-#pragma once
-
-namespace caf {
-
-/// A C++11/14 drop-in replacement for C++17's `std::byte`.
-enum class byte : uint8_t {};
-
-template <class IntegerType,
-          class = std::enable_if_t<std::is_integral<IntegerType>::value>>
-constexpr IntegerType to_integer(byte x) noexcept {
-  return static_cast<IntegerType>(x);
-}
```

Comment:
Please leave this function in the header (but it should take a `std::byte`) and deprecate it.

Reply:
I'm confused about this. Why leave this function and change it to take `std::byte` when the standard-lib provides it already?

Reply:
Because removing the function outright may break client code. We wish to give them a warning with 0.19, not break their code immediately. 🙂